drivers/cloudreve/util.go

```diff
@@ -49,14 +51,18 @@ func (d *Cloudreve) request(method string, path string, callback base.ReqCallbac
 
 		// 刷新 cookie
 		if r.Code == http.StatusUnauthorized && path != loginPath {
-			err = d.login()
-			if err != nil {
-				return err
+			if d.Username != "" && d.Password != "" {
+				err = d.login()
+				if err != nil {
+					return err
+				}
+				return d.request(method, path, callback, out)
 			}
-			return d.request(method, path, callback, out)
+			// 错误影响正常访问，下线该储存
+			d.GetStorage().SetStatus(fmt.Sprintf("%+v", err.Error()))
+			op.MustSaveDriverStorage(d)
+			return errors.New(r.Msg)
```

Comment:
This line should be in the original position

Reply:
No need to do when cookie expired?

Reply:
if r.Code != 0, we should return an error whether the cookie expired rather than continue

Reply:
Done. I also submitted to frontend repo